### PR TITLE
{tools}[foss/2020a] snakemake v7.18.1 w/ Python 3.8.2

### PR DESCRIPTION
**Pull request by jpecar**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+
+dependencies = [
+    ('Python', '3.8.2'),
+    ('SciPy-bundle', '2020.03', '-Python-3.8.2'),
+    ('IPython', '7.15.0', '-Python-3.8.2'),
+    ('poetry', '1.0.9', '-Python-3.8.2'),
+    ('GitPython', '3.1.9', '-Python-3.8.2'),
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:17]: if you're depending on easyconfigs with a `Python` versionsuffix, then this easyconfig should also have that same suffix

```suggestion
name = 'snakemake'
version = '7.18.1'
versionsuffix = '-Python-%(pyver)s'

homepage = 'https://snakemake.readthedocs.io'
description = "The Snakemake workflow management system is a tool to create reproducible and scalable data analyses."

toolchain = {'name': 'foss', 'version': '2020a'}

dependencies = [
    ('Python', '3.8.2'),
    ('SciPy-bundle', '2020.03', versionsuffix),
    ('IPython', '7.15.0', versionsuffix),
    ('poetry', '1.0.9', versionsuffix),
    ('GitPython', '3.1.9', versionsuffix),
]
```

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    ('MarkupSafe', '2.1.1', {
+        'checksums': ['7f91197cc9e48f989d12e4e6fbc46495c446636dfc81b9ccf50bb0ec74b91d4b'],
+    }),
+    ('Jinja2', '3.1.2', {
+        'checksums': ['31351a702a408a9e7595a8fc6150fc3f43bb6bf7e319770cbc0db9df9437e852'],
+    }),
+    ('PyYAML', '6.0', {
+	'modulename': 'yaml',
+        'checksums': ['68fb519c14306fec9720a2a5b45bc9f0c8d1b9c72adf45c37baedfcd949c35a2'],
+    }),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:81]: is there a strong requirement on `PyYAML > 5.3`? If not, I would add a dependency on `('PyYAML', '5.3')` instead

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    ('reretry', '0.11.1', {
+        'checksums': ['4ae1840ae9e443822bb70543c485bb9c45d1d009e32bd6809f2a9f2839149f5d'],
+    }),
+    ('MarkupSafe', '2.1.1', {
+        'checksums': ['7f91197cc9e48f989d12e4e6fbc46495c446636dfc81b9ccf50bb0ec74b91d4b'],
+    }),
+    ('Jinja2', '3.1.2', {
+        'checksums': ['31351a702a408a9e7595a8fc6150fc3f43bb6bf7e319770cbc0db9df9437e852'],
+    }),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:77]: is there a strong requirement on `Jinja 2 > 2.11.2`? If not, I would add a dependency on `('Jinja2', '2.11.2')` instead

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb (hunk 2 above)
Comment (jpecar) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:81]: Yes, there is hard dependency on PyYAML >= 6.0.

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb (hunk 3 above)
Comment (jpecar) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:77]: Yes, there is hard dependency on Jinja2 >= 3.0.

File: easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb (hunk 1 above)
Comment (jpecar) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb:17]: This makes sense, I did --update-pr with updated eb file.

Comment (jpecar) [reply easybuild/easyconfigs/s/snakemake/snakemake-7.18.1-foss-2020a.eb]: Ok, done. I'm still learning how all this works ... thanks for patience :)

**Review by boegel — CHANGES_REQUESTED** [review]
@jpecar Existing `snakemake` easyconfigs also define `$SNAKEMAKE_LOAD_MODULE`, shouldn't we keep that?

```
# site-specific: the line below is the command for loading this module with the (default) EB module naming scheme
# if necessary, change this command to match your cluster setup
local_snakemake_load = 'module load %(module_name)s'
modextravars = {'SNAKEMAKE_LOAD_MODULE': local_snakemake_load}
```

**Comment by boegel** [comment]
> > @jpecar Existing `snakemake` easyconfigs also define `$SNAKEMAKE_LOAD_MODULE`, shouldn't we keep that?
> > ```
> > # site-specific: the line below is the command for loading this module with the (default) EB module naming scheme
> > # if necessary, change this command to match your cluster setup
> > local_snakemake_load = 'module load %(module_name)s'
> > modextravars = {'SNAKEMAKE_LOAD_MODULE': local_snakemake_load}
> > ```
> 
> I've checked our eb archive and modules and we're not using this. Up to you ...

Since the existing `snakemake` easyconfigs have it, we should keep it?

**Comment by smoors** [comment]
> > > @jpecar Existing `snakemake` easyconfigs also define `$SNAKEMAKE_LOAD_MODULE`, shouldn't we keep that?
> > > ```
> > > # site-specific: the line below is the command for loading this module with the (default) EB module naming scheme
> > > # if necessary, change this command to match your cluster setup
> > > local_snakemake_load = 'module load %(module_name)s'
> > > modextravars = {'SNAKEMAKE_LOAD_MODULE': local_snakemake_load}
> > > ```
> > 
> > 
> > I've checked our eb archive and modules and we're not using this. Up to you ...
> 
> Since the existing `snakemake` easyconfigs have it, we should keep it?

yes, please keep these 2 lines in, but maybe commented out. we need this in our cluster because the environment is automatically reset in batch jobs.
please also add back `snakemake-6.6.1_fix_jobs.patch`, which doesn't do anything if `SNAKEMAKE_LOAD_MODULE` is not defined.

**Comment by boegel** [comment]
closing this since `foss/2020a` is deprecated, see https://docs.easybuild.io/policies/toolchains

**Pull request closed without merging**